File: adapter-butterknife/src/main/java/com/azazellj/recyclerview/adapter/butterknife/ButterKnifeViewHolder.java
package com.azazellj.recyclerview.adapter.butterknife;

import android.support.annotation.NonNull;
import android.view.View;

import butterknife.ButterKnife;

/**
 * Created by azazellj on 3/7/18.
 */
public class ButterKnifeViewHolder extends com.azazellj.recyclerview.adapter.common.AdapterViewHolder {
    public ButterKnifeViewHolder(View itemView) {
        super(itemView);

        bindView(this, itemView);
    }

    public static void bindView(@NonNull Object target, @NonNull View source) {
        ButterKnife.bind(target, source);
    }
}

File: adapter-butterknife-kt/src/main/java/com/azazellj/recyclerview/adapter/butterknife/kt/ButterKnifeViewHolder.kt
package com.azazellj.recyclerview.adapter.butterknife.kt

import android.view.View
import butterknife.ButterKnife

/**
 * Created by azazellj on 3/7/18.
 */
class ButterKnifeViewHolder
/**
 * Default constructor.
 *
 * @param itemView view
 */(iteView: View) : com.azazellj.recyclerview.adapter.kt.common.AdapterViewHolder(iteView) {

    init {
        bindView(this, iteView)
    }

    companion object {
        fun bindView(target: Any, source: View) {
            ButterKnife.bind(target, source)
        }
    }
}